kgforge/specializations/models/rdf/store_service.py

```diff
@@ -28,37 +31,43 @@
 
 class StoreService(RdfService):
 
-    def __init__(self, default_store: Store, context_iri: Optional[str] = None,
-                 context_store: Optional[Store] = None) -> None:
+    def __init__(
+        self,
+        default_store: Store,
+        context_iri: Optional[str] = None,
+        context_store: Optional[Store] = None,
+    ) -> None:
 
         self.default_store = default_store
         self.context_store = context_store or default_store
         # FIXME: define a store independent strategy
-        self.NXV = Namespace(self.default_store.service.namespace) if hasattr(self.default_store.service, "namespace") \
-            else Namespace(Service.NEXUS_NAMESPACE_FALLBACK)
-        self.store_metadata_iri = self.default_store.service.store_context if hasattr(self.default_store.service, "store_context") \
+        self.store_metadata_iri = (
+            self.default_store.service.store_context
+            if hasattr(self.default_store.service, "store_context")
             else Namespace(Service.NEXUS_CONTEXT_FALLBACK)
-        self._shapes_to_resources: Dict
-        self._imported = []
-        self._graph = Graph()
+        )
+        self._graph = rdflib.Dataset()
         self._sg = ShapesGraphWrapper(self._graph)
         super().__init__(self._graph, context_iri)
 
-    def schema_source_id(self, schema_iri: str) -> str:
-        return self._shapes_to_resources[schema_iri]
+    def schema_source_id(self, shape_uri: str) -> str:
+        return str(self.shape_to_defining_resource[URIRef(shape_uri)])
+
+    def schema_id(self, shape_uri: str) -> str:
+        return self.schema_source_id(shape_uri)
```

Comment:
Schema_id is the defining resource in both store and directory for now but the schema source id is different: 
- directory: file path
- store: schema_id+"/graph" to avoid named graph and schema to have the same id in the RDFDataset
So both services can't share the same implementation of schema_source_id. For schema_id, they can share it for now.

Reply:
It is the source, and if the source if defined by shape_to_named_graph, then what will tell the implementations apart is how shape_to_named_graph was build = construction, but not which dictionary should be accessed?

Reply:
We could also consider that the source is the delta endpoint that was targeted to retrieve the schema

Reply:
I think if the delta endpoint option isn't deemed appropriate, there should be a comment in _build_shapes_map of store_service.py explaining the reason for /graph because it's not self-explanatory